src/System.AppContext/src/System/AppContext.cs

```diff
@@ -18,8 +18,7 @@ private enum SwitchValueState
             UnknownValue = 0x8 // Not used on .NET Native
         }
 
-        private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
-        private static readonly object s_syncLock = new object();
+        private static readonly Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
```

Comment:
For one thing it allows for using the more natural phrasing in the name of the switch whether the default option is `false` or `true`. `DoTheNewThing` or `DoTheOldThing` depending on which is more easily named rather than `DontDoTheNewThing` or `DontDoTheOldThing`.